icu4j/main/core/src/test/java/com/ibm/icu/dev/test/message2/Sources.java

```diff
@@ -0,0 +1,28 @@
+// © 2024 and later: Unicode, Inc. and others.
+// License & terms of use: https://www.unicode.org/copyright.html
+
+package com.ibm.icu.dev.test.message2;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+
+// Class corresponding to the json test files.
+// See Unit.java and StringToListAdapter.java for how this is used.
+// Workaround for not being able to get the class of a generic type.
+
+class Sources {
+    final List<String> sources;
+
+    Sources(List<String> sources) {
+        this.sources = sources;
+    }
+
+    @Override
+    public String toString() {
+        StringJoiner result = new StringJoiner(", ", "[", "]");
+        result.add(sources.toString());
```

Comment:
This is basically "{" + sources + "}", there is no need for a StringJoiner.

The sources.toString() is just one string.
So the ", " separator in StringJoiner does not matter.
It would, if the code would be something like 
`for (String s : sources) { result.add(s); }`

Reply:
Done in [3c0aa5f](https://github.com/unicode-org/icu/pull/3063/commits/3c0aa5ffa53b92e45f3845c7c69951c318f7ba54)